sys/include/net/sock/util.h

```diff
@@ -268,6 +273,27 @@ static inline bool sock_udp_ep_equal(const sock_udp_ep_t *a,
     return sock_tl_ep_equal(a, b);
 }
 
+/**
+ * @brief   Helper function to establish a DTLS connection
+ *
+ * @param[out]  sock_udp     Struct to store the underlying UDP socket
+ * @param[out]  sock_dtls    Struct for the actual DTLS socket
+ * @param[out]  session      Struct to store DTLS session information
+ * @param[in]   tag          Credential tag to use
+ * @param[in]   local        Local endpoint, must not be NULL
+ * @param[in]   remote       Server endpoint to connect to
+ * @param[in]   work_buf     Buffer used to negotiate connection
+ * @param[in]   work_buf_len Size of @p work buf. Should be at least
+ *                           160 bytes for AES_128_CCM_8 with PSK
+ *
+ * @return  0 on success
+ * @return negative error otherwise (see @ref sock_dtls_recv_aux)
+ */
+int sock_dtls_connect(sock_udp_t *sock_udp, sock_dtls_t *sock_dtls,
+                      sock_dtls_session_t *session, credman_tag_t tag,
+                      sock_udp_ep_t *local, const sock_udp_ep_t *remote,
+                      void *work_buf, size_t work_buf_len);
```

Comment:
I would rather call it `establish_session` than `connect` or something, as DTLS is by definition connection-less (same goes for the "DTLS connection" above).

Reply:
fine with me